main.tf

```
@@ -249,11 +249,20 @@ resource "aws_rds_cluster" "secondary" {
   }
 }
 
+resource "random_pet" "instance" {
+  count  = local.enabled ? 1 : 0
+  prefix = var.cluster_identifier == "" ? module.this.id : var.cluster_identifier
+  keepers = {
+    cluster_family = var.cluster_family
+    instance_class = var.serverlessv2_scaling_configuration != null ? "db.serverless" : var.instance_type
+  }
+}
+
 resource "aws_rds_cluster_instance" "default" {
   count                                 = local.cluster_instance_count
-  identifier                            = var.cluster_identifier == "" ? "${module.this.id}-${count.index + 1}" : "${var.cluster_identifier}-${count.index + 1}"
+  identifier                            = "${random_pet.instance[0].id}-${count.index + 1}"
   cluster_identifier                    = coalesce(join("", aws_rds_cluster.primary[*].id), join("", aws_rds_cluster.secondary[*].id))
-  instance_class                        = var.serverlessv2_scaling_configuration != null ? "db.serverless" : var.instance_type
+  instance_class                        = random_pet.instance[0].keepers.instance_class
```

Typically we try to avoid hard coding `[0]`, as if the resource is enabled (count = 1) it is valid, but if enabled = false, it becomes invalid.

We usually prefer to write this as 
```suggestion
one(random_pet.instance[*].keepers.instance_class)
```

This uses the first index if it exists otherwise sets it to null.

In this case, it ultimately doesn't make a difference, because if `enabled = false` then this resource (`aws_rds_cluster_instance`) won't be created. Though we might change it in a future PR for consistency with other modules

---

Actually, using `one()` remains important, because during `terraform destroy` you can get into a situation where the resource has been destroyed even though `enabled == true`. I forget the exact prerequisites for this kind of occurrence, but we have seen it in practice.